screeps-game-api/src/objects/mod.rs

```diff
@@ -289,6 +291,15 @@ pub unsafe trait CanStoreEnergy: StructureProperties {
     }
 }
 
+/// Used to specify which structures can use their stored energy for spawning creeps.
+///
+/// # Contract
+///
+/// The reference returned from `AsRef<Reference>::as_ref` must be have an
+/// `energy` and an `energyCapacity` properties. Also, that energy can be used
+/// by a spawner to create a new creep.
```

Comment:
Small nit: this fact should already be implied by the thing implementing the contract for `CanStoreEnergy`. Would this structure be something like "... must be accepted by `energyStructures` option in `StructureSpawn.spawnCreep`"?

Reply:
Aren't those two formulation equivalent? I can remove the part about `energy` and `energyCapacity` though.

Reply:
`spawnCreep` specifically checks that the structures in `energyStructures` are `StructureSpawn` or `StructureExtension`: https://github.com/screeps/engine/blob/9d9c6165a48efd31ee3cb87730fb5092a3b10b9f/src/game/structures.js#L1119 - it won't accept just any object with `energy` and `energyCapacity`.

`HasEnergy` on the other hand just requires that it wraps some JS object which has `energy` and `energyCapacity`, right? `StructureLab` is `HasEnergy` and it has `energy` and `energyCapacity` fields, but it wouldn't be valid to implement `HasEnergyForSpawn` for it since `spawnCreep` would not accept it.

Reply:
oh, nevermind, you did have that in there. My apologies I completely missed the "also, that energy can be used by a spawner to create a new creep".